misc/version_updater.py

```diff
@@ -0,0 +1,325 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
+
+"""
+Helper that help changing the version of a tool across the repository.
+"""
+
+import enum
+import re
+import sys
+from argparse import ArgumentParser
+from dataclasses import dataclass
+from fileinput import FileInput
+from pathlib import Path
+from typing import Callable, Dict, List, Tuple, Union
+
+ReplacementPattern = Union[str, Callable[[str], str]]
+
+RawRegexes = List["Replace"]
+
+ROOT_DIR = (Path(__file__) / "../..").resolve()
+
+
+@dataclass
+class Replace:
+    regex: str
+    replaced: ReplacementPattern
+
+    def compile(self, version: str) -> Tuple[re.Pattern[str], str]:
+        regex = re.compile(self.regex)
+        replace = (
+            self.replaced.format(version=version)
+            if isinstance(self.replaced, str)
+            else self.replaced(version)
+        )
+
+        return (regex, replace)
+
+
+def hide_patch_version(template: str, separator: str = ".") -> Callable[[str], str]:
+    def _hide_patch_version(version: str) -> str:
+        return template.format(version=separator.join(version.split(".")[:-1]))
+
+    return _hide_patch_version
+
+
+@enum.unique
+class Tool(enum.Enum):
+    Rust = "rust"
+    Python = "python"
+    Poetry = "poetry"
+    Node = "node"
+    WasmPack = "wasm-pack"
+
+
+TOOLS_VERSION: Dict[Tool, str] = {
+    Tool.Rust: "1.68.0",
+    Tool.Python: "3.9.5",
+    Tool.Poetry: "1.3.2",
+    Tool.Node: "18.12.0",
+    Tool.WasmPack: "0.11.0",
+}
+
+
```

Comment:
There is a lot of duplication in the `FILES_WITH_VERSION_INFO configuration dictionary, this might be error-prone in the future. 

How about defining, whenever possible, specific instances of the replace class?
E.g:

```python
@dataclass
class PythonShortReplace(Replace):
    regex: str = r"python\d.\d+"
    replaced: ReplacementPattern = hide_patch_version("python{version}")
```